### PR TITLE
Check DDR status in order to run

### PR DESCRIPTION
**Pull request by michael-richey**
### What does this PR do?
Check the DDR status at both the source and the destination and don't run if we cannot confirm that it is safe to do so.

### Description of the Change
Calls the endpoint to get the DDR status for both source and destination.
If we cannot get the DDR status exit.
If the status in not ONBOARDING, PASSIVE, or RECOVERY then exit.
The user can override this behavior with `--verify-ddr-status=False`

### Possible Drawbacks

Users that currently use the tool for migration but don't have DDR will need to pass the  `--verify-ddr-status=False` flag

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -148,4 +148,4 @@ dmypy.json
 cython_debug/
 
 # sync-cli resources
-resources/
+/resources/
```
Comment (michael-richey) [inline .gitignore:151]: This rule was ignoring too much.

File: datadog_sync/utils/configuration.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 from __future__ import annotations
 import logging
-from sys import exit
+import sys
```
Comment (michael-richey) [inline datadog_sync/utils/configuration.py:8]: This change was due to a code quality violation from the static analysis bot.

File: README.md [hunk 3]
```diff
@@ -198,6 +201,11 @@ The tools `sync` command provides a cleanup flag (`--cleanup`). Passing the clea
 
 For example, `ResourceA` and `ResourceB` are imported and synced, followed by deleting `ResourceA` from the source organization. Running the `import` command will update the source organizations state file to only include `ResourceB`. The following `sync --cleanup=Force` command will now delete `ResourceA` from the destination organization.
 
+#### Verify DDR status flag
+
+By default all commands now check the Datadog Disaster Recovery (DDR) status of both the source and destination organizations before running. This behavior is controlled by the boolean flag `--verify-ddr-status` or the enviornment variable `DD_VERIFY_DDR_STATUS`.
```
Comment (jhgilbert) [inline README.md:206]:
```suggestion
By default all commands check the Datadog Disaster Recovery (DDR) status of both the source and destination organizations before running. This behavior is controlled by the boolean flag `--verify-ddr-status` or the environment variable `DD_VERIFY_DDR_STATUS`. 
```

**Review by jhgilbert — DISMISSED** [review]
Approved with suggestions for style and to fix a typo. Thanks!

File: datadog_sync/utils/configuration.py [hunk 4]
```diff
@@ -61,14 +63,36 @@ async def init_async(self, cmd: Command):
...
+                    sys.exit(1)
             self.logger.info("clients validated successfully")
 
+        # Don't sync if DDR is active
+        if self.verify_ddr_status:
+            try:
+                await _verify_ddr_status(self.destination_client)
+            except Exception as err:
+                self.logger.error(
+                    "The destination DDR verification failed. Use the --verify-ddr-status "
```
Comment (heyronhay) [inline datadog_sync/utils/configuration.py:80]: Maybe something a bit more descriptive? Like, "The destination DDR verification failed indicating a disaster recovery is in progress"?

File: datadog_sync/utils/configuration.py [hunk 5]
```diff
@@ -61,14 +63,36 @@ async def init_async(self, cmd: Command):
...
+                    f"flag to override, exiting: {err}"
+                )
+                sys.exit(1)
+            try:
+                await _verify_ddr_status(self.source_client)
+            except Exception as err:
+                self.logger.error(
+                    "The source DDR verification failed. Use the --verify-ddr-status "
```
Comment (heyronhay) [inline datadog_sync/utils/configuration.py:88]: More descriptive, echoing whatever we decide for the destination message

File: datadog_sync/utils/configuration.py (hunk 4 above)
Comment (michael-richey) [reply datadog_sync/utils/configuration.py]: Previous error messages:
```
The destination DDR verification failed. Use the --verify-ddr-status flag to override, exiting: Could not get DDR status
The destination DDR verification failed. Use the --verify-ddr-status flag to override, exiting: DDR status: ACTIVE
The destination DDR verification failed. Use the --verify-ddr-status flag to override, exiting: DDR status: FAILOVER
```

New error messages:
```
The destination DDR verification failed. This indicates that no DDR status could be retrieved. Use the --verify-ddr-status flag to override.
The destination DDR verification failed. This indicates disaster recovery is in progress. DDR status retrieved: ACTIVE  
The destination DDR verification failed. This indicates disaster recovery is in progress. DDR status retrieved: FAILOVER
```

File: datadog_sync/utils/configuration.py (hunk 5 above)
Comment (michael-richey) [reply datadog_sync/utils/configuration.py]: Previous error messages:
```
The source DDR verification failed. Use the --verify-ddr-status flag to override, exiting: Could not get DDR status     
The source DDR verification failed. Use the --verify-ddr-status flag to override, exiting: DDR status: ACTIVE           
The source DDR verification failed. Use the --verify-ddr-status flag to override, exiting: DDR status: FAILOVER         
```

New error messages:
```
The source DDR verification failed. This indicates that no DDR status could be retrieved. Use the --verify-ddr-status flag to override.
The source DDR verification failed. This indicates disaster recovery is in progress. DDR status retrieved: ACTIVE
The source DDR verification failed. This indicates disaster recovery is in progress. DDR status retrieved: FAILOVER ```

**Review by heyronhay — APPROVED** [review]

**Pull request merged**